tests/cbor_test.c

```diff
@@ -0,0 +1,532 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/cbor.h>
+#include <float.h>
+#include <math.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+#define CBOR_TEST_CASE(NAME)                                                                                           \
+    AWS_TEST_CASE(NAME, s_test_##NAME);                                                                                \
+    static int s_test_##NAME(struct aws_allocator *allocator, void *ctx)
+
+CBOR_TEST_CASE(cbor_encode_decode_int_test) {
+    (void)allocator;
+    (void)ctx;
+    aws_common_library_init(allocator);
+    enum { VALUE_NUM = 5 };
+
+    /**
+     * Less than 24 only take 1 byte,
+     * 24 to uint8_t max takes 2 bytes
+     * uint8_t max to uint16_t max takes 3 bytes
+     * uint16_t max to uint32_t maxx takes 5 bytes
+     * uint32_t max to uint64_t max takes 9 bytes
+     */
+    uint64_t values[VALUE_NUM] = {23, 24, UINT8_MAX + 1, UINT16_MAX + 1U, UINT32_MAX + 1LLU};
+    uint64_t expected_encoded_len[VALUE_NUM] = {1, 2, 3, 5, 9};
+
+    size_t encoded_len = 0;
+    struct aws_cbor_encoder *encoder = aws_cbor_encoder_new(allocator, 128);
+    /* Unsigned int */
+    for (size_t i = 0; i < VALUE_NUM; i++) {
```

Comment:
nit: we have ARRAYSIZE_OF macro somewhere that you can use to avoid having a constant

Reply:
yeah, arguable which is better as here, we have multiple array that I want to make sure they all the same size